src/typechecker.cpp

```diff
@@ -2083,11 +2097,12 @@ bool TypeChecker::checkExpression(expression_t expr)
             handleError(expr, "$First_argument_of_inline_if_must_be_an_integer");
             return false;
         }
-        if (!areInlineIfCompatible(expr.get_type(), expr[1].get_type(), expr[2].get_type())) {
+
+        type = getInlineIfCommonType(expr[1].get_type(), expr[2].get_type());
+        if (!areInlineIfCompatible(type, expr[1].get_type(), expr[2].get_type())) {
```

Comment:
It should be possible to convert the common type to the resulting type
```suggestion
        if (!areAssignmentCompatible(resultType, type)) {
```

Reply:
I'm not sure what `resultType` is supposed to be. But `areInlineIfCompatible` call does `areAssignmentCompatible` checks

Reply:
given the expression `res = (c) ? (a) : (b)` then the `resultType` is the type of `res`.